app/services/designation_and_undesignation/base.rb

```diff
@@ -0,0 +1,62 @@
+module DesignationAndUndesignation
+  class Base
+    attr_accessor :order_id, :package, :quantity, :is_new_orders_package
+
+    def designate_stockit_item
+      package.designate_to_stockit_order(order_id)
+    end
+
+    def designated_orders_packages
+      package.get_designated_orders_packages
+    end
+
+    def dispatched_location_id
+      id ||= Location.dispatch_location.id
+    end
+
+    def dispatched_orders_packages
+      package.get_dispatched_orders_packages
+    end
+
+    def initialize(package, order_id, quantity, *args)
```

Comment:
Could you `initialize` this at the top of the class - it's one of the most descriptive methods for answering the question "How do I use this class"?